### PR TITLE
Feature/thermostat

### PR DESCRIPTION
**Pull request by hez**
What is working:
- Pairing
- Getting state from the service and displaying it in the Home app

## Expected behaviour

Using this prototype virtual thermostat: https://github.com/hez/example_hap_thermostat

When I pair this thermostat with Apple Home it should allow me to turn on and off the heat via the Home app.

What actually happens:

Apple Home sends a request back to the service with a path of `/prepare` which is not implemented in HAP.

### CODE REVIEW

**Comment by mtrudel** [comment]
I've been waiting a looooong time for this! https://github.com/mtrudel/talks/blob/main/2020-01-Toronto-Elixir-Night-Thousand-Island.pdf

**Comment by mtrudel** [comment]
What's the status of this? By appearances this seems like it ought to work

**Comment by hez** [comment]
It is mostly working, I have created a sample virtual thermostat which can be booted up and paired to show what is not working right now.

**Comment by mtrudel** [comment]
Huh, it looks like HAP doesn't implement section 6.7.2.4 "Timed Write Procedures" in the spec. I don't recall ever seeing this before; I think it must have fallen through the cracks. I'll get into this over the holidays!

**Comment by mtrudel** [comment]
Give this a rip - I tested it locally with your demo project and it seems to work as expected.

File: lib/hap/encrypted_http_server.ex [hunk 1]
```diff
@@ -82,6 +82,13 @@ defmodule HAP.EncryptedHTTPServer do
     end
   end
 
+  # Note that we do not enforce timed writes per 6.7.2.4; we just throw this request on the ground
+  put "/prepare" do
+    conn
+    |> put_resp_header("content-type", "application/hap+json")
+    |> send_resp(200, Jason.encode!(%{status: 0}))
```
Comment (hez) [inline lib/hap/encrypted_http_server.ex:89]: hah! thats what that need to be... I had tried something very similar but not exactly the same... 🤦 


I'll give this a shot!

**Comment by hez** [comment]
Gave this a good going over and appears to work well!

Tested the following
✅  push/pull current_state
✅  push/pull target_temp
✅  push current_temp and current_humidity

**Comment by mtrudel** [comment]
Looks good! Pending green CI I'll merge this right away

**Pull request merged**